src/components/accounts/accounts.js

```diff
@@ -104,8 +109,35 @@ export class AccountsComponent {
     console.log("handleEditClick");
   }
 
-  handleDeleteClick() {
-    console.log("handleDeleteClick");
+  handleDeleteClick(event) {
+    let moreButton = event.target.closest(".button-more");
+    this.listItem = moreButton.closest(".accounts__list-item");
+    this.confirmDialog.showDialog("account", this.listItem.innerText);
+  }
+
+  initConfirmDialog() {
+    this.confirmDialog = new ConfirmDialogComponent(
+      this.confirmDialogMountPoint,
+      {
+        onOkClick: this.handleDeleteConfirm.bind(this)
+      }
+    );
+    this.confirmDialog.mount();
+  }
+
+  handleDeleteConfirm() {
+    this.delAccount(this.listItem);
+  }
+
+  delAccount(listItem) {
+    let index = Array.from(this.accountsList.children).indexOf(listItem);
+    AccountListService.del(index).then(() => {
+      this.accountsList.removeChild(listItem);
+    });
+    let accountId = listItem.dataset.id;
```

Comment:
please first delete all transactions for account and then delete account -- and chain these promises